extensions/websocket/02_websocketstream.js

```diff
@@ -0,0 +1,403 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+"use strict";
+
+/// <reference path="../../core/internal.d.ts" />
+
+((window) => {
+  const core = window.Deno.core;
+  const webidl = window.__bootstrap.webidl;
+  const { writableStreamClose, Deferred } = window.__bootstrap.streams;
+  const { DOMException } = window.__bootstrap.domException;
+  const {
+    StringPrototypeEndsWith,
+    StringPrototypeToLowerCase,
+    Symbol,
+    SymbolFor,
+    Set,
+    ArrayPrototypeMap,
+    ArrayPrototypeJoin,
+    PromisePrototypeThen,
+    PromisePrototypeCatch,
+    Uint8Array,
+    TypeError,
+  } = window.__bootstrap.primordials;
+
+  webidl.converters.WebSocketStreamOptions = webidl.createDictionaryConverter(
+    "WebSocketStreamOptions",
+    [
+      {
+        key: "protocols",
+        converter: webidl.converters["sequence<USVString>"],
+        defaultValue: [],
```

Comment:
```suggestion
        get defaultValue() {
          return [];
        },
```

Otherwise the _exact_ same array will be used each time (identity would be the same).